lib/galaxy/celery/tasks.py

```diff
@@ -389,3 +393,48 @@ def prune_history_audit_table(sa_session: galaxy_scoped_session):
 def cleanup_short_term_storage(storage_monitor: ShortTermStorageMonitor):
     """Cleanup short term storage."""
     storage_monitor.cleanup()
+
+
+@galaxy_task(action="clean up job working directories")
+def cleanup_jwds(sa_session: galaxy_scoped_session, object_store: BaseObjectStore):
+    """Cleanup job working directories for failed jobs that are older than X days"""
+
+    def get_failed_jobs():
+        failed_jobs = {}
+        jobs = sa_session.query(model.Job).filter(
+            model.Job.state == "error",
```

Comment:
it might make sense to check against all terminal states here (not in `ok`, `queued`, `running`, `new`), but maybe there's a better way to do that. I know that @natefoo and I occasionally manually fail jobs with a state liked `manually_failed` and it'd be good to make sure we're getting *all* terminal jobs.

Reply:
I think we can tweak this after the first version is merged.

Reply:
works for me